src/vsr/replica.zig

```diff
@@ -6969,10 +6954,8 @@ pub fn ReplicaType(
         /// Replication to standbys works similarly, jumping off the replica just before primary.
         /// TODO Use recent heartbeat data for next replica to leapfrog if faulty (optimization).
         fn replicate(self: *Self, message: *Message.Prepare) void {
-            assert(self.status == .normal);
             assert(message.header.command == .prepare);
-            assert(message.header.view == self.view);
-            assert(message.header.op == self.op);
+            assert(message.header.view >= self.view);
```

Comment:
Let's add a `maybe(message.header.op < self.op)`. Hm, actually, maybe we could assert implication here?

```zig
if (message.header.op < self.op) assert(message.header.view < self.view)
```

In general, it's always usetful to think which asserts are missing!

Reply:
Ack, makes sense! Adding `maybe(message.header.op < self.op)` since by this point we've already dropped older prepares so we can't `assert(message.header.view < self.view)`. The older ops that we forward may be either from a primary of a newer view, or from a primary from a primary of the same view whose start_view we're waiting on (https://github.com/tigerbeetle/tigerbeetle/pull/2143#discussion_r1693251200).

Reply:
https://github.com/tigerbeetle/tigerbeetle/pull/2143/commits/f6895cea32ffe0e02e969be767bc8f9057d2e1c3